packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts

```diff
@@ -57,7 +63,18 @@ export abstract class BaseDeployExecutor extends SfdxCommandletExecutor<
     });
 
     execution.processExitSubject.subscribe(async exitCode => {
-      this.logMetric(execution.command.logName, startTime);
+      let properties;
+      const registryAccess = new RegistryAccess();
+      try {
+        const components = registryAccess.getComponentsFromPath(
+          execFilePathOrPaths
+        );
+        const metadataCount = JSON.stringify(createComponentCount(components));
+        properties = { metadataCount };
+        this.logMetric(execution.command.logName, startTime, properties);
```

Comment:
This needs to be moved below the try/catch block. If there's an error, we won't log anything at all.